### PR TITLE
"Starts_With" should not be marked as inline, and write_component_static_forwarder should be.

### PR DESCRIPTION
**Pull request by pi1024e**
The reason for this comes down to clang's interpretation. Starts_with is already entirely in a struct, while "write_component_static_forwarder" is not, and should be marked as such.

### CODE REVIEW

File: cppwinrt/component_writers.h [hunk 1]
```diff
@@ -289,7 +289,7 @@ catch (...) { return winrt::to_hresult(); }
             bind<write_consume_args>(signature));
     }
 
-    void write_component_static_forwarder(writer& w, MethodDef const& method)
+    inline void write_component_static_forwarder(writer& w, MethodDef const& method)
```
Comment (kennykerr) [inline cppwinrt/component_writers.h]: This should be static not inline. Look at the rest of the functions in this source file. This helps the compiler issue a warning if the function is unused.

File: cppwinrt/helpers.h [hunk 2]
```diff
@@ -384,7 +384,6 @@ namespace cppwinrt
             info.overridable = overridable || has_attribute(impl, "Windows.Foundation.Metadata", "OverridableAttribute");
             info.base = base;
             info.generic_param_stack = generic_param_stack;
-            writer::generic_param_guard guard;
```
Comment (kennykerr) [inline cppwinrt/helpers.h]: Why did you move this? This is an unrelated to change to inlining.

File: cppwinrt/helpers.h [hunk 3]
```diff
@@ -498,7 +497,7 @@ namespace cppwinrt
         return result;
     }
 
-    bool has_fastabi_tearoffs(writer& w, TypeDef const& type)
+    inline bool has_fastabi_tearoffs(writer& w, TypeDef const& type)
```
Comment (kennykerr) [inline cppwinrt/helpers.h]: These again should be `static` not `inline`.

File: cppwinrt/type_writers.h [hunk 4]
```diff
@@ -44,12 +44,12 @@ namespace cppwinrt
         }
     };
 
-    bool operator==(type_name const& left, type_name const& right)
+    inline bool operator==(type_name const& left, type_name const& right)
```
Comment (kennykerr) [inline cppwinrt/type_writers.h]: These don't strictly need to be inline because they're only ever included in a single TU. And again, static will issue a compiler warning if unused.

File: strings/base_extern.h [hunk 5]
```diff
@@ -1,8 +1,8 @@
 
-__declspec(selectany) int32_t(__stdcall* winrt_to_hresult_handler)(void* address) noexcept {};
-__declspec(selectany) void(__stdcall* winrt_suspend_handler)(void const* token) noexcept {};
-__declspec(selectany) void(__stdcall* winrt_resume_handler)(void const* token) noexcept {};
-__declspec(selectany) int32_t(__stdcall* winrt_activation_handler)(void* classId, winrt::guid const& iid, void** factory) noexcept {};
+inline __declspec(selectany) int32_t(__stdcall* winrt_to_hresult_handler)(void* address) noexcept {};
```
Comment (kennykerr) [inline strings/base_extern.h]: Why did this need to be inline? These are used internally and have very specific linkage requirements and I'd rather not mess with them.

File: cppwinrt/type_writers.h (hunk 4 above)
Comment (pi1024e) [reply cppwinrt/type_writers.h]: For all the inlines? or just that one?

File: strings/base_events.h [hunk 6]
```diff
@@ -248,7 +248,6 @@ namespace winrt::impl
                 ((*reinterpret_cast<impl::abi_t<I>**>(&object))->*(Method))(m_token);
             }
         }
-    private:
```
Comment (kennykerr) [inline strings/base_events.h]: Empty line here.

File: strings/base_version.h [hunk 7]
```diff
@@ -3,7 +3,7 @@
 
 // WINRT_version is used by Microsoft to analyze C++/WinRT library adoption and inform future product decisions.
 extern "C"
-__declspec(selectany)
+inline __declspec(selectany)
```
Comment (kennykerr) [inline strings/base_version.h]: Why is this necessary?

File: cppwinrt/helpers.h [hunk 8]
```diff
@@ -498,7 +498,7 @@ namespace cppwinrt
         return result;
     }
 
-    bool has_fastabi_tearoffs(writer& w, TypeDef const& type)
+    staitc bool has_fastabi_tearoffs(writer& w, TypeDef const& type)
```
Comment (kennykerr) [inline cppwinrt/helpers.h]: Typo.

File: strings/base_version.h (hunk 7 above)
Comment (pi1024e) [reply strings/base_version.h]: Because it is in a struct

Comment (pi1024e) [reply strings/base_version.h]: The compiler is implicitly in-lining this anyway.

Comment (kennykerr) [reply strings/base_version.h]: Then leave it alone. Unless it's causing compilation issues, this has worked as-is for a few years and I don't want to disrupt that unnecessarily.

File: cppwinrt/component_writers.h [hunk 9]
```diff
@@ -1208,4 +1208,4 @@ namespace winrt::@::implementation
                 slot);
         }
     }
-}
```
Comment (kennykerr) [inline cppwinrt/component_writers.h]: Please leave a new line at the end of each file. Some compilers don't like it otherwise.
